### PR TITLE
Refer issue #12, issue #20, issue #32

### PR DESCRIPTION
**Pull request by nguyenvo8**
Added Apache Derby Library
Enabled option -Xlint: all -Weberror
Resolved the warnings

### CODE REVIEW

**Comment by nguyenvo8** [comment]
I updated my fork before started working.  When I run my build.gradle file, it gave back all check style error.  I thought that somebody that finished changing all the style errors. Should I fix it by myself, because it seems like other PRs work fine? @rsanchez-wsu would you help me out on this? Thank you

**Comment by Aashishkebab** [comment]
You made some changes that caused checkstyle issues. There is a threshold that is allowed. If you make another violation, then the threshold is passed. 

You must fix your violations. You can find your changes in the log by doing Ctrl + F to find your file in the log, and see which lines have violations that you modified.

**Comment by nguyenvo8** [comment]
@Aashishkebab Thank you, but where should I find the log? Can you be more details?

**Comment by Aashishkebab** [comment]
Click on the "details" button next to the Travis build, or click on one of the red Xs.

**Comment by nguyenvo8** [comment]
Thank you

**Comment by nguyenvo8** [comment]
@rsanchez-wsu I was about to work on resolving the conflict, but I couldn't see where the conflict occurred. How should I handle this?

File: Test/Controller/DataControllerTest.java [hunk 1]
```diff
@@ -31,29 +31,39 @@
...
-    @Test
-    public void validNodeList() throws Exception
-    {
-    }
-
-    @Test
-    public void loadHubFile() throws Exception
-    {
-    }
+	/** set up method.
```
Comment (rsanchez-wsu) [inline Test/Controller/DataControllerTest.java]: This Javadoc is not useful.  This method, like others, is empty.  While we certainly want to correct the style problems and make sure that everything is properly documented, adding documentation that does not add value is not a good strategy.  Additionally, the formatting of this JavaDoc is not consistent with the JavaDoc we have elsewhere.  A better JavaDoc comment would be:
```
/**
 * Run prior to each unit test to set up the data structures and environment needed by the test.
 *
 * @throws Exception if there are any problems with the test setup, indicating that there is a problem to investigate
 */
```

Also note that JUnit has annotations for methods that run prior each individual unit test method and for methods that run only once prior to every test method in a given test class.  The same is true for the `@After` and related annotations.  You will need to make sure that your documentation accurately reflects whether the method will run before/after each test or before/after all of the tests.

As far as formatting, you of course need to make sure that the comment begins on the correct line (yours is one line too high) and that proper sentence capitalization is used.

Everything I have pointed out here applies for just about every JavaDoc.

**Pull request closed without merging**